src/components/templates/Camp/components/Map.tsx

```diff
@@ -479,29 +491,36 @@ export const Map: React.FC<PropsType> = ({
                 }}
               >
                 <div
-                  className={`camp-venue ${
-                    roomClicked === room.title ? "clicked" : ""
-                  }`}
+                  className={`camp-venue ${roomClicked === room.title ? "clicked" : ""
+                    }`}
                 >
                   <div
-                    className={`grid-room-btn ${
-                      isUnderneathRoom && "isUnderneath"
-                    }`}
+                    className={`grid-room-btn ${isUnderneathRoom && "isUnderneath"
+                      }`}
                   >
-                    <div
-                      onClick={(e) => {
-                        e.stopPropagation();
-                        const isExternalUrl = isExternalLink(room.url);
-                        window.open(
-                          getRoomUrl(room.url),
-                          isExternalUrl ? "_blank" : "noopener,noreferrer"
-                        );
-                        roomEnter(room);
-                      }}
-                      className="btn btn-white btn-small btn-block"
-                    >
-                      Join now
-                    </div>
+                    {isExternalLink(room.url) ? (
+                      <div
+                        className="btn btn-white btn-small btn-block"
+                        onClick={(e) => {
```

Comment:
The code in onClick is repeated on line #515, it would make sense to make that a method which gets called onClick instead of 3 functions being called separately